sdk/formrecognizer/azure-ai-formrecognizer/tests/test_document.py

```diff
@@ -0,0 +1,158 @@
+
+        # check page range
+        assert len(raw_analyze_result.pages) == len(returned_model.pages)
+
+    @FormRecognizerPreparer()
+    @DocumentAnalysisClientPreparer()
+    def test_document_multipage_transform(self, client):
+        with open(self.multipage_invoice_pdf, "rb") as fd:
+            document = fd.read()
+
+        responses = []
+
+        def callback(raw_response, _, headers):
+            analyze_result = client._deserialize(AnalyzeResultOperation, raw_response)
+            extracted_document = AnalyzeResult._from_generated(analyze_result.analyze_result)
+            responses.append(analyze_result)
+            responses.append(extracted_document)
+
+        poller = client.begin_analyze_document("prebuilt-document", document, cls=callback)
+        result = poller.result()
+        raw_analyze_result = responses[0].analyze_result
+        returned_model = responses[1]
+
+        # Check AnalyzeResult
+        assert returned_model.model_id == raw_analyze_result.model_id
+        assert returned_model.api_version == raw_analyze_result.api_version
+        assert returned_model.content == raw_analyze_result.content
+        
+        self.assertDocumentPagesTransformCorrect(returned_model.pages, raw_analyze_result.pages)
+        self.assertDocumentTransformCorrect(returned_model.documents, raw_analyze_result.documents)
+        self.assertDocumentTablesTransformCorrect(returned_model.tables, raw_analyze_result.tables)
+        self.assertDocumentKeyValuePairsTransformCorrect(returned_model.key_value_pairs, raw_analyze_result.key_value_pairs)
+        self.assertDocumentEntitiesTransformCorrect(returned_model.entities, raw_analyze_result.entities)
+        self.assertDocumentStylesTransformCorrect(returned_model.styles, raw_analyze_result.styles)
+
+        # check page range
+        assert len(raw_analyze_result.pages) == len(returned_model.pages)
+
+    @FormRecognizerPreparer()
+    @DocumentAnalysisClientPreparer()
+    def test_document_multipage_table_span_pdf(self, client):
+        with open(self.multipage_table_pdf, "rb") as fd:
+            myfile = fd.read()
+        poller = client.begin_analyze_document("prebuilt-document", myfile)
+        document = poller.result()
+        assert len(document.tables) == 3
+        assert document.tables[0].row_count == 29
+        assert document.tables[0].column_count == 5
+        assert document.tables[1].row_count == 6
+        assert document.tables[1].column_count == 4
+        assert document.tables[2].row_count == 23
+        assert document.tables[2].column_count == 5
+
+    @FormRecognizerPreparer()
+    @DocumentAnalysisClientPreparer()
+    def test_document_specify_pages(self, client):
+        with open(self.multipage_invoice_pdf, "rb") as fd:
+            document = fd.read()
+
+        poller = client.begin_analyze_document("prebuilt-document", document, pages=["1"])
```

Comment:
should not be list of pages? but maybe this merged before the swagger PR?

Reply:
This ended up getting fixed in a separate PR.